build_tools/circle/build_doc.sh

```diff
@@ -58,32 +58,45 @@ get_build_type() {
         return
     fi
     changed_examples=$(echo "$filenames" | grep -E "^examples/(.*/)*plot_")
-    if [ -n "$filenames" ]
-    then
-        for af in ${filenames[@]}
-        do
-            if [ $af != "build_tools/circle/build_doc.sh" ]
-            then
-                page_figure=$(grep figure $af | grep auto_example | awk -F "/" '{print $NF}' | sed 's/sphx_glr_//' | awk -F "_" '{OFS="_";$NF=""; print $0}')
-                page_image=$(grep image $af | grep auto_example | awk -F "/" '{print $NF}' | sed 's/sphx_glr_//' | awk -F "_" '{OFS="_";$NF=""; print $0}')
-            fi
-        done
-    fi
-    if [ -n "$page_figure" ]
-    then
-        changed_examples+=$(echo ${page_figure::-1} | sed 's/_ /|/g')
-    fi
-    if [ -n "$page_image" ]
+
+    if [[ -n "$filenames" ]]
     then
-        changed_examples+=$(echo ${page_image::-1} | sed 's/_ /|/g')
+        # get rst files
+        rst_files="$(echo "$filenames" | grep -E "rst$")"
+
+        # get lines with figure or images
+        img_fig_lines="$(echo "$rst_files" | xargs grep -shE "(figure|image)::")"
+
+        # get only auto_examples
+        auto_example_files="$(echo "$img_fig_lines" | grep auto_examples | awk -F "/" '{print $NF}')"
+
+        # remove "sphx_glr_" from path and accept replace _\d\d\d.png with .py
+        image_paths="$(echo "$auto_example_files" | sed 's/sphx_glr_//' | sed -e 's/_[[:digit:]][[:digit:]][[:digit:]].png/.py/')"
```

Comment:
Looks like the thumb images are still here, we need to add another filter for py files:

```bash
echo "$auto_example_files" | sed 's/sphx_glr_//' | sed -e 's/_[[:digit:]][[:digit:]][[:digit:]].png/.py/' | grep -sh ".py$"
```

Reply:
Could you please me explain what are thumb images? Thanks ! :)
(I mean, you probably mean thumbnails somewhere, but I don't know examples of that)

Reply:
Sphinx gallery generates files like `IMAGE_NAME_001.png` and `IMAGE_NAME_thumb.png` where the thumbs are used for thumbnails.

I was running some of these commands and saw some appear here.

Reply:
And where do `_thumb.png` are linked? Could they be linked anywhere in `.rst` files?